gemini/use-cases/retrieval-augmented-generation/multimodal_rag_langchain.ipynb

```diff
@@ -0,0 +1,1026 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Create [`MatchingEngineIndexEndpoint`](https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.MatchingEngineIndexEndpoint)\n",
+    "  - https://cloud.google.com/vertex-ai/docs/vector-search/deploy-index-public"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DEPLOYED_INDEX_ID = \"mm_rag_langchain_index_endpoint\"\n",
+    "\n",
+    "index_endpoint = aiplatform.MatchingEngineIndexEndpoint.create(\n",
+    "    display_name=DEPLOYED_INDEX_ID,\n",
+    "    description=\"Multimodal RAG LangChain Index Endpoint\",\n",
+    "    public_endpoint_enabled=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Deploy Index to Index Endpoint\n",
+    "  - NOTE: You can stop this cell after starting it instead of waiting for deployment\n",
+    "  - You can check the status at https://console.cloud.google.com/vertex-ai/matching-engine/indexes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "index_endpoint = index_endpoint.deploy_index(\n",
+    "    index=index, deployed_index_id=\"mm_rag_langchain_deployed_index\"\n",
+    ")\n",
+    "index_endpoint.deployed_indexes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create retriever & load documents"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Create [`VectorSearchVectorStore`](https://api.python.langchain.com/en/latest/vectorstores/langchain_google_vertexai.vectorstores.vectorstores.VectorSearchVectorStore.html) with Vector Search Index ID and Endpoint ID.\n",
+    "- Use [`textembedding-gecko`](https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/text-embeddings) as embedding model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The vectorstore to use to index the summaries\n",
+    "vectorstore = VectorSearchVectorStore.from_components(\n",
+    "    project_id=PROJECT_ID,\n",
+    "    region=LOCATION,\n",
+    "    gcs_bucket_name=GCS_BUCKET,\n",
+    "    index_id=\"5333409848946065408\",\n",
+    "    endpoint_id=\"7088687803713716224\",\n",
+    "    embedding=VertexAIEmbeddings(model_name=\"textembedding-gecko@003\"),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Alternatively, use Chroma for a local vector store."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vectorstore = Chroma(\n",
+    "    collection_name=\"mm_rag_test\",\n",
+    "    embedding_function=VertexAIEmbeddings(model_name=\"textembedding-gecko@003\"),\n",
```

Comment:
Suggest commenting this out so that users will only use one defined vector DB?

Reply:
I put it in a different cell so that they could choose to run it or not, but if you think it's more clear to comment it out, I'll do it.